src/Nethermind/Nethermind.Core/Utils/ConcurrentWriteBatcher.cs

```diff
@@ -40,6 +40,24 @@ public void PutSpan(ReadOnlySpan<byte> key, ReadOnlySpan<byte> value, WriteFlags
         ReturnWriteBatch(currentBatch);
     }
 
+    public void Merge(ReadOnlySpan<byte> key, ReadOnlySpan<byte> value, WriteFlags flags = WriteFlags.None)
+    {
+        IWriteBatch currentBatch = RentWriteBatch();
+        currentBatch.Merge(key, value, flags);
+        ReturnWriteBatch(currentBatch);
+    }
+
+    public void Clear()
+    {
+        if (_disposing)
+            throw new InvalidOperationException("Batch is already disposed.");
```

Comment:
Put it into `ThrowXXX` method (avoids JITing until needed and decreases hot-path code size)